api/lib/domain/models/BadgeResults.js

```diff
@@ -0,0 +1,39 @@
+const _ = require('lodash');
+
+class BadgeResults {
+  constructor(badges, skills, knowledgeElements) {
+    this.masteryPercentage = _computeMasteryPercentage(skills, knowledgeElements);
+
+    this.campaignParticipationBadges = badges.map(({ id, badgePartnerCompetences }) => {
+      return {
+        id,
+        partnerCompetenceResults: _buildPartnerCompetenceResults(badgePartnerCompetences, knowledgeElements),
+      };
+    });
+  }
+
+  static build(badges, skills, participationResults) {
```

Comment:
Pour utiliser le même nommage que dans le constructeur:

```suggestion
  static build(badges, skills, knowledgeElements) {
```

Reply:
done